python/ray/cluster_utils.py

```diff
@@ -88,7 +89,11 @@ def start(self, _system_config=None):
         env = os.environ.copy()
         env.update({"AUTOSCALER_UPDATE_INTERVAL_S": "1", "RAY_FAKE_CLUSTER": "1"})
         self._process = subprocess.Popen(cmd, env=env)
```

Comment:
@ericl any reason why we need to use `subprocess.Popen` with `ray start --block` instead  of `subproces.check_call` which will wait until ray start command finishes.

Reply:
It allows us to kill it later with process.kill() below.

Reply:
But if we use `subprocess.check_call` with `ray start` instead of `subprocess.Popen` with `ray start --block`, then we don't need to do `process.kill()`?

Reply:
How do you ensure the particular ray instance is killed then? You have to track the pid. This way is cleaner and self contained in the Process object. Every cluster is one Process.

Reply:
Oh, you are talking about running multiple `AutoscalingCluster` at the same time? Won't `subprocess.check_call(["ray", "stop", "--force"])` just kills everything?